src/components/Other/Project.vue

```diff
@@ -43,25 +80,47 @@
 </template>
 
 <script>
-import { mapGetters } from 'vuex'
+import { mapState, mapActions, mapGetters } from 'vuex'
 
 export default {
   metaInfo: {
     title: 'Project'
   },
-  methods: {
-    ...mapGetters('projects', ['getProjectById'])
-  },
-  computed: {
-    project() {
-      return this.getProjectById()(this.id)
-    }
-  },
   props: ['id'],
   data() {
     return {
       loading: true
     }
+  },
+  computed: {
+    loadedProject() {
+      return this.getProjectById()(this.id)
+    },
+    ...mapState('projects', ['project'])
+  },
+  methods: {
+    ...mapActions('projects', ['setProjectRef']),
+    ...mapGetters('projects', ['getProjectById'])
+  },
+  created() {
+    if (this.loadedProject == null) {
```

Comment:
If condition such that if that getter returns null, then only fetch the project from firebase